pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -343,16 +342,21 @@ class ModelValidatorImpl implements ModelValidator {
                     List<DescribableParameter> requiredParams = model.parameters.findAll { it.isRequired() }
 
                     if (requiredParams.size() != meth.args.size()) {
-                        errorCollector.error(meth, Messages.ModelValidatorImpl_WrongNumberOfStepParameters(meth.name, requiredParams.size(), meth.args.size()))
-                        valid = false
+                        // TODO: This is a specialized hack for allowing single-required-Boolean-parameter constructors
```

Comment:
For me TODOs in code are punch lines to jokes you'll tell a couple years from now. 😄 

Put another way, think of this from the perspective of not-you.  Not-you sees `TODO` but no clear statement of what is _to be done_.  What does that mean?   You know what you mean, but not-you does not. 

I see `TODO` I expect as statement of what is to be done.  
```
//TODO: revisit post 1.0
// This allows single-required-Boolean-parameter constructors
// to be called like "foo()", Groovy-style, passing null as the parameter value.
// <description of why this needs revisiting> 
```